### PR TITLE
docs(firebasejson config): missing docs for FirebaseJsonConfig

### LINKED ISSUE #4626: [📚] Missing FirebaseJsonConfig types
## Documentation Feedback

Documentation about `FirebaseJsonConfig` types is missing [here](https://rnfirebase.io/reference/app/firebasejsonconfig), see the screenshot:

image

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

### PR DESCRIPTION
**Pull request by b3coded**
### Description

See Discussion #4626 for context. 
The FirebaseJsonConfig interface is unused, so we can transform interface FirebaseJsonConfig into a JSON Schema to provide a better documentation about it, Editor Insellisense and maybe some validations.

Also added a page about FirebaseJsonConfig in `Core/App`

### Related issues

Fixes #4626 

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [x] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
🔥

### CODE REVIEW

File: docs/sidebar.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@
   - - - Usage
       - '/firestore/usage'
     - - Usage with Emulator
-      - '/firestore/emulator'
```
Comment (mikehardy) [inline docs/sidebar.yaml:46]: haha - thanks for cleaning up my whitespace error, that was mine...

**Review by mikehardy — APPROVED** [review]
Fairly far-reaching but a real improvement I think. I like this a lot!
I'll click around the rendered preview and then (for my own piece of mind) re-verify the statements in related discussion about how/where it's used, but I can't think of a single reason not to merge at the moment, even the prev/next nav is hooked up, nice job

**Comment by mikehardy** [comment]
Looks like it could use an add of Intellisense to the spellcheck dictionary

**Comment by mikehardy** [comment]
Mmmm - okay - nothing technically wrong - but I feel like clicking on the new sidebar link should show the available options with their descriptions right there. Is there a way to either include the contents into that page during render or to copy/paste them in (with a note in the schema that if you make changes there, you need to paste them into the docs as well? That would be the true holy grail as they would all be documented well on the docs website plus you get intellisense etc

**Comment by b3coded** [comment]
Great idea!
I will create a script for this

**Comment by b3coded** [comment]
@mikehardy Done

**Comment by mikehardy** [comment]
Fantastic! I put it in my queue for a re-look with bias towards merge since it looked really close on first pass already, thanks!

File: website/gatsby-browser.js [hunk 2]
```diff
@@ -17,37 +17,34 @@
...
-    }
-  }
+  // if (window && window.document && window.location && window.location.hash) {
+  //   const { location } = window;
+  //   const el = window.document.getElementById(location.hash.replace('#', ''));
+  //   if (el) {
+  //     const rect = el.getBoundingClientRect();
+  //     window.scrollTo(0, rect.top);
+  //   }
+  // }
```
Comment (mikehardy) [inline website/gatsby-browser.js:27]: This actually fixes an existing bug unrelated to this PR that I've noticed for a while, that the hash navigation doesn't work. Or did not work but it works now on this preview. What happened there and why does this fix it? Just curious

File: website/gatsby-browser.js [hunk 3]
```diff
@@ -17,37 +17,34 @@
...
 
 exports.onPreRouteUpdate = () => {
   // Set the current sidebar scroll height on the window ready for the page change
-  const sidebar = window.document.getElementById('sidebar');
-  window.RNFB_SIDEBAR_SCROLL_POSITION = sidebar ? sidebar.scrollTop : 0;
+  // const sidebar = window.document.getElementById('sidebar');
+  // window.RNFB_SIDEBAR_SCROLL_POSITION = sidebar ? sidebar.scrollTop : 0;
 };
 
 exports.onRouteUpdate = ({ location }) => {
```
Comment (mikehardy) [inline website/gatsby-browser.js:36]: lint error - location is unused now

File: website/scripts/parse-mdx.js [hunk 4]
```diff
@@ -15,11 +15,16 @@
...
  */
 
+
 module.exports = async function parseMdx({ node, getNode, actions }) {
+
   if (node.internal.type !== 'Mdx') return;
 
   // Mdx nodes parents are a File
-  const File = getNode(node.parent);
+  const File = await getNode(node.parent);
```
Comment (mikehardy) [inline website/scripts/parse-mdx.js:24]: if getNode was async before and not used as an await, this looks like a serious issue! where is/was it defined? I'm having trouble with the cross reference

File: website/src/reference/Methods.tsx [hunk 5]
```diff
@@ -63,6 +63,8 @@ function Reference({
 }
 
 function Methods({ methods }: { methods: Method[] }) {
+  console.log(methods);
```
Comment (mikehardy) [inline website/src/reference/Methods.tsx:66]: lint error - no console statement should go live

**Review by mikehardy — CHANGES_REQUESTED** [review]
Sorry for the delay in reviewing, juggling a lot of plates at the moment and took a moment to get back to react-native-firebase

This appears to work in general (which is amazing actually) but I had a couple comments to understand better, and also it seems to have a layout overflow issue on my mac using firefox?

Screen Region 2020-12-10 at 07 16 04

File: website/gatsby-browser.js (hunk 2 above)
Comment (b3coded) [reply website/gatsby-browser.js:27]: This is a default behavior of the browser when using hash navigation. With the current code I couldn't  use the hash .
So I just commented this code to try to fix it  it and it started working

File: website/scripts/parse-mdx.js (hunk 4 above)
Comment (b3coded) [reply website/scripts/parse-mdx.js:24]: I tried without await and it worked, maybe when I edited this file it doesn't generated my components so I thought that getNode was async but was my mistake, fixing it

**Comment by b3coded** [comment]
@mikehardy I tested on Reference API page with long words and it causes a layout overflow too ( you can see this on current [Analytics Reference](https://rnfirebase.io/reference#analytics) ),
so I added `word-break: break-word;` for all `p` tags inside tables in `styles.css` and it fixed the overflow.

File: website/src/reference/Methods.tsx (hunk 5 above)
Comment (b3coded) [reply website/src/reference/Methods.tsx]: removed

File: website/gatsby-browser.js (hunk 3 above)
Comment (b3coded) [reply website/gatsby-browser.js]: removed

**Comment by mikehardy** [comment]
> @mikehardy I tested on Reference API page with long words and it causes a layout overflow too ( you can see this on current [Analytics Reference](https://rnfirebase.io/reference#analytics) ),
> so I added `word-break: break-word;` for all `p` tags inside tables in `styles.css` and it fixed the overflow.

Weird - your change definitely fixed the JSON Config document - so it's a step forward definitely - but I couldn't reproduce an overflow in the browsers I checked for the analytics reference on the currently deployed site. At the same time, with your change I also don't see a regression there, so I don't think it's a step backwards. So I think the word break is effective - a good change

**Comment by b3coded** [comment]
Current Analytics Reference
image
Here is the Analytics  reference overflow, caused by Link without word-break

Analytics reference with my change
image

**Review by mikehardy — APPROVED** [review]
LGreatTM - when I click around in the preview site I don't just have new / correct JSON Config documentation but the # navigation also works, that's a real and separate improvement all by itself. Thank you!

**Pull request merged**